src/routes/v1/secret_test.go

```diff
@@ -11,76 +12,376 @@ import (
+		},
+	}
+	createTestNamespace(secretNamespace)
+	createTestSecret(secretName, secretNamespace)
+
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			request, _ := http.NewRequest("GET", fmt.Sprintf("/v1/namespaces/%s/secrets/%s",
+				test.requestUri.namespace, test.requestUri.secret), nil)
+			writer := httptest.NewRecorder()
+			router.ServeHTTP(writer, request)
+			assert.Equal(t, test.want.statusCode, writer.Code)
+
+			var response map[string]interface{}
+			if err := json.Unmarshal(writer.Body.Bytes(), &response); err != nil {
+				panic(err)
+			}
+			assert.Equal(t, test.want.response, response)
+
+		})
+	}
 }
 
 func TestGetSecrets(t *testing.T) {
-	request, _ := http.NewRequest("GET", "/v1/namespaces/default/secrets/", nil)
-	writer := httptest.NewRecorder()
-	router.ServeHTTP(writer, request)
-
-	assert.Equal(t, http.StatusOK, writer.Code)
-	var response types.GetSecretsResponse
-	err := json.Unmarshal(writer.Body.Bytes(), &response)
-	assert.NoError(t, err)
+	secretNamespace := "test-namespace-secrets"
+	type requestUri struct {
+		namespace string
+	}
+
+	type want struct {
+		statusCode int
+		response   types.GetSecretsResponse
+	}
+
+	cases := map[string]struct {
+		requestUri requestUri
+		want       want
+	}{
+		"ShouldSuccessGettingSecrets": {
+			requestUri: requestUri{
+				namespace: secretNamespace,
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: types.GetSecretsResponse{Count: 2, Secrets: []types.Secret{{SecretName: "test-secret1", NamespaceName: secretNamespace, Type: "Opaque"},
+					{SecretName: "test-secret2", NamespaceName: secretNamespace, Type: "Opaque"}}},
+			},
+		},
+	}
+	createTestNamespace(secretNamespace)
+	createTestSecret("test-secret1", secretNamespace)
+	createTestSecret("test-secret2", secretNamespace)
+
+	for name, test := range cases {
+		t.Run(name, func(t *testing.T) {
+			request, _ := http.NewRequest("GET", fmt.Sprintf("/v1/namespaces/%s/secrets/",
+				test.requestUri.namespace), nil)
+			writer := httptest.NewRecorder()
+			router.ServeHTTP(writer, request)
+			assert.Equal(t, test.want.statusCode, writer.Code)
+
+			var response types.GetSecretsResponse
+			if err := json.Unmarshal(writer.Body.Bytes(), &response); err != nil {
+				panic(err)
+			}
+			assert.Equal(t, test.want.response, response)
+
+		})
+	}
+}
+
+func TestDeleteSecrets(t *testing.T) {
+	secretNamespace := "test-namespace-deleteSecret"
+	secretName := "test-secret"
+	type requestUri struct {
+		namespace string
+		secret    string
+	}
+
+	type want struct {
+		statusCode int
+		response   map[string]string
+	}
+
+	cases := map[string]struct {
+		requestUri requestUri
+		want       want
+	}{
+		"ShouldSuccessGettingSecret": {
+			requestUri: requestUri{
+				secret:    secretName,
+				namespace: secretNamespace,
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response: map[string]string{
+					"message": fmt.Sprintf("Secret %q was deleted successfully", secretName)},
```

Comment:
Use constant like here (move the constant to another file) - https://github.com/dana-team/platform-backend/blob/1989d979499464dac34fd2dbb08b0ea01c502f3a/src/routes/v1/capp_test.go#L23